test/mocking_utils/patch.hpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef MOCKING_UTILS__PATCH_HPP_
+#define MOCKING_UTILS__PATCH_HPP_
+
+#include <functional>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#include "mimick/mimick.h"
+#include "rcutils/macros.h"
+
+namespace mocking_utils
+{
+
+template<size_t N, typename SignatureT>
+struct PatchTraits;
+
+template<size_t N, typename ReturnType>
+struct PatchTraits<N, ReturnType(void)>
+{
+  mmk_mock_define(mock_type, ReturnType);
+};
+
+template<size_t N, typename ReturnType, typename ArgType0>
```

Comment:
It's not because `mmk_mock_define` is a macro that needs all the parameter types given as separate arguments at the preprocessing stage -- it won't play along with parameter packs.